src/main/java/htsjdk/variant/vcf/VCFFileReader.java

```diff
@@ -16,105 +16,275 @@
-	    try (final VCFFileReader vcfFileReader = new VCFFileReader(file, false)) {
-		    return vcfFileReader.getFileHeader().getSequenceDictionary();
-		}
-	}
-
-    /** Constructs a VCFFileReader that requires the index to be present. */
-	public VCFFileReader(final File file) {
-		this(file, true);
-	}
-
-    /** Constructs a VCFFileReader with a specified index. */
+    private final FeatureReader<VariantContext> reader;
+
+    /**
+     * Returns true if the given file appears to be a BCF file.
+     *
+     * @deprecated in favor of {@link VCFFileReader#isBCF(Path)}
+     */
+    @Deprecated
+    public static boolean isBCF(final File file) {
+        return isBCF(file.toPath());
+    }
+
+    /**
+     * Returns true if the given path appears to be a BCF file.
+     */
+    public static boolean isBCF(final Path path) {
+        return path.toUri().getRawPath().endsWith(".bcf");
+    }
+
+    /**
+     * Returns the SAMSequenceDictionary from the provided VCF file.
+     *
+     * @deprecated in favor of {@link VCFFileReader##getSequenceDictionary(Path)}
+     */
+    @Deprecated
+    public static SAMSequenceDictionary getSequenceDictionary(final File file) {
+        try (final VCFFileReader vcfFileReader = new VCFFileReader(file, false)) {
+            return vcfFileReader.getFileHeader().getSequenceDictionary();
+        }
+    }
+
+    /**
+     * Constructs a VCFFileReader that requires the index to be present.
+     *
+     * @deprecated in favor of {@link VCFFileReader#VCFFileReader(Path, boolean)}
+     */
+    @Deprecated
+    public VCFFileReader(final File file) {
+        this(file, true);
+    }
+
+    /**
+     * Constructs a VCFFileReader with a specified index.
+     *
+     * @deprecated in favor of {@link VCFFileReader#VCFFileReader(Path, Path)}
+     */
+    @Deprecated
     public VCFFileReader(final File file, final File indexFile) {
         this(file, indexFile, true);
     }
 
-    /** Allows construction of a VCFFileReader that will or will not assert the presence of an index as desired. */
-	public VCFFileReader(final File file, final boolean requireIndex) {
-	  // Note how we deal with type safety here, just casting to (FeatureCodec)
-	  // in the call to getFeatureReader is not enough for Java 8.
-      FeatureCodec<VariantContext, ?> codec = isBCF(file) ? new BCF2Codec() : new VCFCodec();
-      this.reader = AbstractFeatureReader.getFeatureReader(
-                      file.getAbsolutePath(),
-                      codec,
-                      requireIndex);
-	}
+    /**
+     * Allows construction of a VCFFileReader that will or will not assert the presence of an index as desired.
+     *
+     * @deprecated in favor of {@link VCFFileReader##VCFFileReader(Path, boolean)}
+     */
+    @Deprecated
+    public VCFFileReader(final File file, final boolean requireIndex) {
+        // Note how we deal with type safety here, just casting to (FeatureCodec)
```

Comment:
Maybe it is better to call `this(file.toPath(), requireIndex)` instead of doing the magic in place (it should result in the same reader).

Reply:
done.